python-checks/src/main/java/org/sonar/python/checks/ChildAndParentExceptionCaughtCheck.java

```diff
@@ -57,22 +63,63 @@ private static void checkCaughtExceptions(SubscriptionContext ctx, Map<ClassSymb
     caughtExceptionsBySymbol.forEach((currentSymbol, caughtExceptionsWithSameSymbol) -> {
       Expression currentException = caughtExceptionsWithSameSymbol.get(0);
       if (caughtExceptionsWithSameSymbol.size() > 1) {
-        PreciseIssue issue = ctx.addIssue(currentException, "Remove this duplicate Exception class.");
+        var quickFix = createQuickFix(currentException);
+
+        IssueWithQuickFix issue = (IssueWithQuickFix) ctx.addIssue(currentException, "Remove this duplicate Exception class.");
+        issue.addQuickFix(quickFix);
         caughtExceptionsWithSameSymbol.stream().skip(1).forEach(e -> issue.secondary(e, "Duplicate."));
       }
-      PreciseIssue issue = null;
+      IssueWithQuickFix issue = null;
       for (Map.Entry<ClassSymbol, List<Expression>> otherEntry : caughtExceptionsBySymbol.entrySet()) {
         ClassSymbol comparedSymbol = otherEntry.getKey();
         if (currentSymbol != comparedSymbol && currentSymbol.isOrExtends(comparedSymbol)) {
           if (issue == null) {
-            issue = ctx.addIssue(currentException, "Remove this redundant Exception class; it derives from another which is already caught.");
+            var quickFix = createQuickFix(currentException);
+
+            issue = (IssueWithQuickFix) ctx.addIssue(currentException, "Remove this redundant Exception class; it derives from another which is already caught.");
+            issue.addQuickFix(quickFix);
           }
           addSecondaryLocations(issue, otherEntry.getValue());
         }
       }
     });
   }
 
+  private static PythonQuickFix createQuickFix(Expression currentException) {
+    PythonQuickFix.Builder builder = PythonQuickFix.newQuickFix(QUICK_FIX_MESSAGE);
+
+    var fromLine = currentException.firstToken().line();
+    var fromColumn = currentException.firstToken().column();
+    var toLine = currentException.lastToken().line();
+    var toColumn = currentException.lastToken().column();
+
+    if (currentException.parent().is(Tree.Kind.TUPLE)) {
+      var parentTuple = (Tuple) currentException.parent();
+      var currentIndex = parentTuple.children().indexOf(currentException);
+
+      // If currentException is not first one - need to remove previous comma
+      if (currentIndex > 1) {
+        var previous = parentTuple.children().get(currentIndex - 1);
+        var previousToken = previous.lastToken();
+        fromLine = previousToken.line();
+        fromColumn = previousToken.column();
+      }
+
+      var nextIndex = currentIndex + 1;
+      // If currentException is not last one - need to remove next comma
+      if (currentIndex < parentTuple.children().size() - 2) {
```

Comment:
What if there's a trailing comma in the tuple?
Like `except (RuntimeException, NotImplementedError,)`
Probably a point can be made that we want to keep it (and it seems that we currently do), I just want to make sure this is intentional (probably we can use a test for this).
If we want to remove such trailing comma, maybe using the elements directly could also be a bit easier here as well.

Reply:
test added